### PR TITLE
Lobby Player Limit

### PR DESCRIPTION
**Pull request by SaddestUnderscore**
This adds a lobby limit as an option that is configurable in the Lobby select screen
- Within a minimum of 2 and a maximum of 32 players
- Adds a little text where it also tells the max of the lobby in the bottom right of lobby cards

### CODE REVIEW

**Comment by 6fears7** [comment]
Once you believe you're good to go to have it tested, please set the PR to ready for review.

**Comment by SaddestUnderscore** [comment]
Hopefully this should be good to go, slight disclaimers though:
- testing in local won't do anything since this is more meant for online, plus the fact that i can't get it to work in local anyway
- i made the check for playercount to accept 1 player and no more, if the test is successful i will put the limit to 4 which is the normal

File: Online/Matchmaking/SteamMatchmakingManager.cs [hunk 1]
```diff
@@ -120,7 +121,24 @@ public override void CreateLobby(LobbyVisibility visibility, string gameMode)
 
         public override void JoinLobby(LobbyInfo lobby)
         {
-            m_JoinLobbyCall.Set(SteamMatchmaking.JoinLobby(lobby.id));
+            if (lobby.mode == "Story")
+            {
+                if (lobby.playerCount > 1)
```
Comment (Chrometurtle) [inline Online/Matchmaking/SteamMatchmakingManager.cs:126]: The limit should be set from a UI element, not hardcoded

Comment (SaddestUnderscore) [reply Online/Matchmaking/SteamMatchmakingManager.cs:126]: yeah you are right, i do want to attempt making the limit configurable, but for now it's hardcoded for the meantime (until i figure out making it configurable)

File: Online/Matchmaking/SteamMatchmakingManager.cs [hunk 2]
```diff
@@ -120,7 +121,24 @@ public override void CreateLobby(LobbyVisibility visibility, string gameMode)
 
         public override void JoinLobby(LobbyInfo lobby)
         {
-            m_JoinLobbyCall.Set(SteamMatchmaking.JoinLobby(lobby.id));
+            if (lobby.mode == "Story")
```
[the remark this answers is not in the document]
Comment (SaddestUnderscore) [reply Online/Matchmaking/SteamMatchmakingManager.cs:124]: like arena and freeroam right

**Comment by 6fears7** [comment]
We can using a Debug build whenever you're free next.

**Comment by SaddestUnderscore** [comment]
i'll list whats changed in this commit: added an option for the player to set the player max in the form of textbox, currently doesn't do anything yet, moved meadow display to the bottom so it doesn't interfere with the coords once everything is done and merged (including passwords where coords), scratched lobby mode detection

File: Menu/LobbySelectMenu.cs [hunk 3]
```diff
@@ -89,6 +87,20 @@ public LobbySelectMenu(ProcessManager manager) : base(manager, RainMeadow.Ext_Pr
...
+            where.y -= 45;
+            var limitNumberLabel = new ProperlyAlignedMenuLabel(this, mainPage, Translate("Player max:"), where, new Vector2(400, 20f), false, null);
+            mainPage.subObjects.Add(limitNumberLabel);
+            where.y -= 5;
+            lobbyLimitNumberTextBox = new OpTextBox(new Configurable<int>(maxPlayerLimit), where, 160f);
+            new UIelementWrapper(this.tabWrapper, lobbyLimitNumberTextBox);
+            where.y += 5; 
+
+            // display version
+            where = new Vector2(0f, 0f);
```
Comment (Chrometurtle) [inline Menu/LobbySelectMenu.cs:100]: Bad, don't do this

**Comment by SaddestUnderscore** [comment]
Okay, after lots of pain and debugging, this _Should_ work, I think I'm actually ready now for making this ready for review

**Comment by 6fears7** [comment]
@Chrometurtle : 

Functionally, this is ready to go. When a lobby is full, the user is prompted, denied entry, and will safely return back to the main menu and can make their own game. 

When lobby has room, will join.

Picture of popup prompt:
image


Please check the code logic at your earliest convenience. In terms of stability, it seems warranted to merge to main.

File: Online/Resource/Lobby.cs [hunk 4]
```diff
@@ -15,10 +15,10 @@ public class Lobby : OnlineResource
 
         public string[] mods = RainMeadowModManager.GetActiveMods();
         public static bool modsChecked;
-
+        public int maxPlayerCount;
```
Comment (Chrometurtle) [inline Online/Resource/Lobby.cs:18]: Looks like everything is already self contained in LobbyInfo. So I don't think this isn't needed anymore

File: Online/Matchmaking/SteamMatchmakingManager.cs [hunk 5]
```diff
@@ -8,6 +8,7 @@ namespace RainMeadow
 {
     public class SteamMatchmakingManager : MatchmakingManager
     {
+        public static int maxLobbyCount;
```
Comment (Chrometurtle) [inline Online/Matchmaking/SteamMatchmakingManager.cs:11]: Same with this guy

File: Online/Matchmaking/MatchmakingManager.cs [hunk 6]
```diff
@@ -13,6 +13,7 @@ public abstract class MatchmakingManager
         public static string NAME_KEY = "name";
         public static string MODE_KEY = "mode";
         public static string PASSWORD_KEY = "password";
+        public static int MAX_LOBBY = 4;
```
[the remark this answers is not in the document]
Comment (SaddestUnderscore) [reply Online/Matchmaking/MatchmakingManager.cs:16]: MAX_LOBBY is needed, maxLobbyCount no

**Comment by Chrometurtle** [comment]
lgtm, there is a way to simplify MAX_LOBBY ;3 but otherwise looks good. Merging

**Pull request merged**